Sources/MongoSwift/MongoCollection+FindAndModify.swift

```diff
@@ -147,4 +212,100 @@ public struct FindOneAndUpdateOptions: Encodable {
 
     /// An optional `WriteConcern` to use for the command.
     public let writeConcern: WriteConcern?
+
+    fileprivate func asOpts() throws -> FindAndModifyOptions {
+        return try FindAndModifyOptions(arrayFilters: arrayFilters, bypassDocumentValidation: bypassDocumentValidation,
+                                        collation: collation, maxTimeMS: maxTimeMS, projection: projection,
+                                        returnDocument: returnDocument, sort: sort, upsert: upsert,
+                                        writeConcern: writeConcern)
+    }
+
+    /// Convenience initializer allowing any/all parameters to be omitted/optional
+    public init(arrayFilters: [Document]? = nil, bypassDocumentValidation: Bool? = nil, collation: Document? = nil,
+                maxTimeMS: Int64? = nil, projection: Document? = nil, returnDocument: ReturnDocument? = nil,
+                sort: Document? = nil, upsert: Bool? = nil, writeConcern: WriteConcern? = nil) {
+        self.arrayFilters = arrayFilters
+        self.bypassDocumentValidation = bypassDocumentValidation
+        self.collation = collation
+        self.maxTimeMS = maxTimeMS
+        self.projection = projection
+        self.returnDocument = returnDocument
+        self.sort = sort
+        self.upsert = upsert
+        self.writeConcern = writeConcern
+    }
+}
+
+/// A class wrapping a `mongoc_find_and_modify_opts_t`, for use with `MongoCollection.findAndModify`
+private class FindAndModifyOptions {
+    // an `OpaquePointer` to a `mongoc_find_and_modify_opts_t`
+    var _options: OpaquePointer?
+
+    init() {
+        self._options = mongoc_find_and_modify_opts_new()
+    }
+
+    // swiftlint:disable:next cyclomatic_complexity
+    init(arrayFilters: [Document]? = nil, bypassDocumentValidation: Bool? = nil, collation: Document?,
+         maxTimeMS: Int64?, projection: Document?, remove: Bool? = nil, returnDocument: ReturnDocument? = nil,
+         sort: Document?, upsert: Bool? = nil, writeConcern: WriteConcern?) throws {
+        self._options = mongoc_find_and_modify_opts_new()
+
+        if let bypass = bypassDocumentValidation,
+        !mongoc_find_and_modify_opts_set_bypass_document_validation(self._options, bypass) {
+            throw MongoError.invalidArgument(message: "Error setting bypassDocumentValidation to \(bypass)")
```

Comment:
I don't really know if these errors are necessary, which is why I have held off on documenting `invalidArgument` errors being thrown from the API methods. 
looking at the source code, all these setters only return false if the provided value is nil, which we can guarantee is never going to happen since we unwrap values first.

Reply:
> all these setters only return false if the provided value is nil, 

Are you referring to the `BSON_ASSERT()` on the `options` struct? I believe that either aborts or is compiled out in release builds. Otherwise, [this setter](https://github.com/mongodb/mongo-c-driver/blob/1.11.0/src/libmongoc/src/mongoc/mongoc-find-and-modify.c#L153) should always return `true`.

Reply:
Unless you were referring to the [field](https://github.com/mongodb/mongo-c-driver/blob/1.11.0/src/libmongoc/src/mongoc/mongoc-find-and-modify.c#L104) and general [append](https://github.com/mongodb/mongo-c-driver/blob/1.11.0/src/libmongoc/src/mongoc/mongoc-find-and-modify.c#L194) setters. I don't see the harm in being defensive and checking for errors, though. We generally do this in PHPC when appending options, even though things are only liable to fall if memory cannot be allocated (in which case we have bigger problems).

Reply:
yeah, I was thinking of the ones for fields, sort, etc. 
but sounds good, let's be defensive, and we can also do the maxTimeMS > 0 check. and then document that an `invalidArgument` error is thrown if any options are invalid.